### DISCUSSION TITLE
How to show the spinner whilst a coroutine runs.

### DISCUSSION THREAD

**Discussion by Aikufurr** [zauberzeug/nicegui#2670] [Q&A]
### Question

My goal is for the spinner to show, the coroutine runs the time intensive task, then when the data returns, delete the spinner and show the output. Currently on the button press it hangs then shows the contents from show_output(), if an error occurs with inner() then the spinner is shown, so it's being deleted before it's shown in normal use. TIA

I've tried the following:
```py
await inner()
await asyncio.get_event_loop().create_task(inner())
await asyncio.get_running_loop().create_task(inner())
```

Code Snippet:
```py
async def execute():
    error_lines.clear()
    warning_lines.clear()
    
    spinner = ui.spinner('gears', size='8em')  

    async def inner():
        if paths["log"]:
            with open(paths["log"], "r") as file:
                await check_lines_for_keywords(file)
        else:
            logs = await get_systemd_logs(inp_unit.value)
            await check_lines_for_keywords(logs)
  
    await asyncio.get_event_loop().create_task(inner())
        
    spinner.delete()
    show_output.refresh(first_run=False)

show_output(first_run=True)
btn_analyse.on('click', execute)
```

**Comment by falkoschindler** [comment]
Hi @Aikufurr,

I don't immediately see the problem. In principle, this should work:

```py
async def run():
    spinner = ui.spinner()
    await asyncio.sleep(1)
    spinner.delete()

ui.button('Run', on_click=run)
```

Since your code is incomplete, I can't try it out myself. You can try creating a minimum reproducible example to better isolate the problem.
